packages/server/lib/makeDataContext.ts

```diff
@@ -119,6 +107,8 @@ export function makeDataContext (options: MakeDataContextOptions): DataContext {
         return cache.removeAllProjectPreferences()
       },
       insertProjectPreferencesToCache (projectTitle: string, preferences: Preferences) {
+        // FIXME: this should be awaited (since it writes to disk asynchronously) but is not
+        // https://cypress-io.atlassian.net/browse/UNIFY-1705
```

Comment:
Any reason we can't just fix that now? Looks like you should just need a `return cache.` here and a type change in `ProjectApiShape` to `Promise<void>`, and an `await` in `setProjectPreferences`

Reply:
Edit: I see the comment you had in the jira, makes sense